server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -264,7 +264,7 @@ private Delta(final long step, final long deltaColumnOffset,
                 final BitSet subscribedColumns, final BitSet modifiedColumns) {
             this.step = step;
             this.deltaColumnOffset = deltaColumnOffset;
-            this.update = TableUpdateImpl.copy(update);
+            this.update = TableUpdateImpl.copy(update); // TODO: can this be acquire()
```

Comment:
No. The delta is queued and processed asynchronously w.r.t. the update graph's cycle. MCS are often re-used, and so you need to ensure that you have your own copy of the MCS.
```suggestion
            this.update = TableUpdateImpl.copy(update);
```